DXRVanilla/DeusEx/Classes/BlackHelicopter.uc

```diff
@@ -58,11 +58,12 @@ function Scream()
 
 function Destroyed()
 {
-    //Daddy screamed real good before he died!
-    Scream();
-
-    //Also generate gibs
-    ChunkUp();
+    if (class'DXRando'.default.dxr.localURL == "15_AREA51_BUNKER") {
```

Comment:
I really don't like the idea of a class checking what level it's in before doing something.

I can't think of a much better solution, although checking for MS_BeginSabotage flag might be slightly better.

I think the main issue is that the Mission15 script doesn't destroy him in an actual damage based way and just uses Destroy, which makes it hard to distinguish.  There *are* two versions of Jock in A51, so theoretically we could have it check for his tag (heli_sabotaged), but that doesn't feel particularly good either.

Reply:
Tag is probably the best way

Reply:
Wait how does the vanilla code know whether to spawn metal fragments or not?

Reply:
All the fragments and explosion graphics are just generated in the mission script before hitting the chopper with Destroy.  I'd be vaguely more tempted to have Fixups enable the "death" functionality on that chopper on first entry rather than putting map specific logic into the class code

Reply:
Yeah, I haven't been seeing any metal fragments in these inappropriate Jock deaths.